lib/screens/action/action_flow.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:provenance_dart/wallet_connect.dart';
+import 'package:provenance_wallet/common/flow_base.dart';
+import 'package:provenance_wallet/common/pw_design.dart';
+import 'package:provenance_wallet/common/widgets/modal/pw_modal_screen.dart';
+import 'package:provenance_wallet/screens/action/action_list/action_list_bloc.dart';
+import 'package:provenance_wallet/screens/action/action_list/action_list_screen.dart';
+import 'package:provenance_wallet/screens/transaction/transaction_confirm_screen.dart';
+import 'package:provenance_wallet/services/models/account.dart';
+import 'package:provenance_wallet/services/models/requests/send_request.dart';
+import 'package:provenance_wallet/services/models/requests/sign_request.dart';
+import 'package:provenance_wallet/services/models/wallet_connect_session_request_data.dart';
+import 'package:provenance_wallet/util/assets.dart';
+import 'package:provenance_wallet/util/get.dart';
+import 'package:provenance_wallet/util/messages/message_field_name.dart';
+import 'package:provenance_wallet/util/strings.dart';
+
+class ActionFlow extends FlowBase {
+  const ActionFlow({required this.account, Key? key}) : super(key: key);
+
+  final Account account;
+
+  @override
+  State<StatefulWidget> createState() => ActionFlowState();
+}
+
+class ActionFlowState extends FlowBaseState implements ActionListNavigator {
+  @override
+  void initState() {
+    super.initState();
+
+    get.pushNewScope(scopeName: "ActionScope");
```

Comment:
For this to be the case we would need to define something 5 screens deep into a flow, but then reference it at the root. Also maybe we are putting too much into get_it. Shouldn't be used to app wide services?

Reply:
I would argue blocs should not even be in it since they are so tied to the related screen

Reply:
> For this to be the case we would need to define something 5 screens deep into a flow, but then reference it at the root. Also maybe we are putting too much into get_it. Shouldn't be used to app wide services?

The gist in the link above has an example

Reply:
i am all for passing in blocs instead of placing them into get_it. An alternative is to use a Provider widget. That would also give use automatic disposal when the widget gets popped off the stack

Reply:
> defining the scope at the flow level

The scope is created at the flow level, but it affects the app globally. So issues could arise later when unrelated code is written.

Reply:
Using `Provider` or passing in seems fine

Reply:
I will switch for this flow and then we can evaluate for others.

Reply:
replaced the get_it code with a provider. This should address all concerns.